### PR TITLE
#463 Restructuring of documentation TOC

### PR DESCRIPTION
**Pull request by jrudolph**
Refs #436 

The idea is to start restructuring the top-level TOC to give the most important topics the best spots. The "chapter numbers" are mostly a cosmetic feature that feels to give the docs an ordered appearance (and hopefully prevents adding more entries lightly). I also added some other smaller "features".

screenshot from 2017-12-22 16 07 19

### CODE REVIEW

File: docs/src/main/paradox/scala/http/06extensions.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# 6. Extensions ➕
```
Comment (2m) [inline docs/src/main/paradox/scala/http/06extensions.md]: The extensions icon does not render here. Neither Chromium, nor IntelliJ. That is a rare plus you found. :)

Nevertheless, I like the icons. :)

File: docs/src/main/paradox/scala/http/06extensions.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# 6. Extensions
+
+This page list well-known third-part extensions of Akka HTTP. See the Akka HTTP tag on scaladex for more info. // TODO: add link
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/06extensions.md]: Link to https://akka.io/community/  

or https://index.scala-lang.org/search?topics=akka-http

**Comment by ktoso** [comment]
Release notes last? Or do something so they don't continue growing into forever in the TOC?

**Comment by ktoso** [comment]
Not sure the utf characters make sense though hm, looks a bit silly

**Comment by jrudolph** [comment]
> Release notes last? Or do something so they don't continue growing into forever in the TOC?

Yep, I guess at some point we should fold them into subpages like "10.0.x release note" etc.

> Not sure the utf characters make sense though hm, looks a bit silly

I removed them again for now. We can still add them later on if we (unlikely enough) happen to agree on something. ;)

**Comment by ktoso** [comment]
> I removed them again for now. We can still add them later on if we (unlikely enough) happen to agree on something. ;) 

ok, sorry ;-)

**Comment by jonas** [comment]
I'm not sure about the Dingbats. It's an extremely limited "icon set" and many of the proposed icons give no clue at all, which IMO makes them pointless. For example, security often uses a shield icon, extensions/plugins a jigsaw puzzle piece.

**Comment by jrudolph** [comment]
@jonas, I see. Let's not go there for now.

File: docs/src/main/paradox/scala/http/routing-dsl/index.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# High-level Server-Side API
+# Routing DSL
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/index.md]: Oh yeah great rename

File: docs/src/main/paradox/scala/http/server-side/low-level-api.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-# Low-Level Server-Side API
+# Core Server API
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/server-side/low-level-api.md]: Also good rename, I always thought not many people get the "low / high level" wording we used

File: docs/src/main/paradox/scala/http/security.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# Security Announcements
+# ! Security Announcements !
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/security.md]: good ;) I can live with that ;)

**Comment by jrudolph** [comment]
A few more changes here:

 * removed the numbering of top-level files
 * copied contributing.md into contributing section, I think it's nice to have that inside the documentation (not sure how to handle that duplication, probably we could link to the documentation from the contributing.md)
 * A bit of page title cleanup (Server-side API => Server API, Low-level Server-Side API => Core Server API, High-level server-side API => Routing DSL)
 * Filled out technologies page
 * Moved some sections under "Tips And Tricks"
 * Reference section that links to reference information at the bottom

**Comment by jrudolph** [comment]
Solved conflicts.

**Pull request merged**